File: Q11/main.cpp
#include<iostream>
using namespace std;

int main(){
    int n,target;
    cin>>n;
    int arr[n];
    //input will be a sorted array
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    cin>>target;

    int start=0,end=n-1;

    while(start<end){
        int sum=arr[start]+arr[end];    
        if(sum==target){
            cout<<arr[start]<<" "<<arr[end];
            return 0;
        }
        else if(sum<target)
            start++;
        else
            end--;
    }
    return 0;
}

File: Q11/main.py
n=int(input())
arr=[]
for i in range(n):
    arr.append(int(input()))

target = int(input())

start,end=0,n-1

while(start<end):
    sum=arr[start]+arr[end]

    if sum==target:
        print(arr[start],arr[end])
        break
    elif sum<target:
        start+=1
    else:
        end-=1